src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -705,25 +705,25 @@ XamlObjectDefinition topLevelControl
 
 		private void BuildSingleTimeInitializer(IIndentedStringBuilder writer, string propertyType, string propertyName, Action propertyBodyBuilder, bool isStatic = true)
 		{
+			// The property type may be partially qualified, try resolving it through FindType
+			var propertySymbol = FindType(propertyType);
+			propertyType = propertySymbol?.GetFullName() ?? propertyType;
+
 			var sanitizedPropertyName = SanitizeResourceName(propertyName);
 
 			var propertyInitializedVariable = "_{0}Initialized".InvariantCultureFormat(sanitizedPropertyName);
 			var backingFieldVariable = "__{0}BackingField".InvariantCultureFormat(sanitizedPropertyName);
 			var staticModifier = isStatic ? "static" : "";
 			var publicPropertyType = FindType(propertyType)?.DeclaredAccessibility == Accessibility.Public
-				? GetGlobalizedTypeName(propertyType)
-				: "object";
-
-			// The property type may be partially qualified, try resolving it through FindType
-			var propertySymbol = FindType(propertyType);
-			propertyType = propertySymbol?.GetFullName() ?? propertyType;
+				? propertyType
+				: "System.Object";
 
 			writer.AppendLineInvariant($"private {staticModifier} bool {propertyInitializedVariable} = false;");
 			writer.AppendLineInvariant($"private {staticModifier} {GetGlobalizedTypeName(propertyType)} {backingFieldVariable};");
```

Comment:
Nit:
- We could remove `GetGlobalizedTypeName` from here and make it part of `var propertySymbol = FindType(propertyType);` aka https://github.com/unoplatform/uno/pull/1278/files#diff-96c40c2bc19dfed15a00fa81b15461d7R708